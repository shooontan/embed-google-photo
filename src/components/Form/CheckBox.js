// @flow
import * as React from 'react';

type Props = {
  label?: string,
};

const CheckBox = (props: Props) => {
  const { label } = props;
  return (
    <p>
      <input type="checkbox" {...props} />
      {label}
    </p>
  );
};

CheckBox.defaultProps = {
  label: '',
};

export default CheckBox;
