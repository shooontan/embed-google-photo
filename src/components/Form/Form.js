// @flow
import * as React from 'react';

type Props = {
  value?: string,
  onChange?: Function,
};

const Form = (props: Props) => {
  const { value, onChange } = props;
  return <input value={value} onChange={onChange} />;
};

Form.defaultProps = {
  value: '',
  onChange: () => {},
};

export default Form;
