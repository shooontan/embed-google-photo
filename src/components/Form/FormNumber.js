// @flow
import * as React from 'react';

type Props = {
  label?: string,
};

const FormNumber = (props: Props) => {
  const { label } = props;
  return (
    <p>
      <input type="number" {...props} />
      {label}
    </p>
  );
};

FormNumber.defaultProps = {
  label: '',
};

export default FormNumber;
