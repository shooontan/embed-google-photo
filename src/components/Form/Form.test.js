// @flow
import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Form from './Form';

Enzyme.configure({ adapter: new Adapter() });

test('Form Component', () => {
  let testRenderer = TestRenderer.create(<Form />);
  let testTree = testRenderer.toJSON();
  expect(testTree).toMatchSnapshot();

  testRenderer = TestRenderer.create(<Form value="test form" />);
  testTree = testRenderer.toJSON();
  expect(testTree).toMatchSnapshot();

  const mock = jest.fn();
  testRenderer = shallow(<Form onChange={mock} />);
  testRenderer.find('input').simulate('change');
  expect(mock).toHaveBeenCalled();
});
