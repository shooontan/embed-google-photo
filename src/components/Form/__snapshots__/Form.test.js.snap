// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component 1`] = `
<input
  onChange={[Function]}
  value=""
/>
`;

exports[`Form Component 2`] = `
<input
  onChange={[Function]}
  value="test form"
/>
`;
